clang/lib/Sema/TreeTransform.h

```diff
@@ -2896,6 +2896,9 @@ class TreeTransform {
     SS.Adopt(QualifierLoc);
 
     Base = BaseResult.get();
+    if (Base->containsErrors())
+      return ExprError();
```

Comment:
I am still traveling back from St Louis, so I can't double check this:

Why is this point the farthest we can go if Base contains errors?

It seems we can build a reference expression over it from the cases above, but why not below?

Reply:
@mizvekov The problems arise when we build a `MemberExpr` for a class member access expression with an object expression that isn't dependent (as far as clang is concerned), but is _indirectly_ dependent (i.e. may instantiate to an expression of a different type). If the object expression is transformed into a `RecoveryExpr`, we end up with a `MemberExpr` that has the type `ASTContext::DependentTy` -- which we _could_ recover from by rebuilding it as a `RecoveryExpr` (or `CXXDependentScopeMemberExpr`), but the simplest solution is to simply bail as we do in practically every other case when part of the class member access expression instantiates to something invalid.